src/user/user.controller.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { BadRequestException, NotFoundException } from "@nestjs/common";
+import { Test, TestingModule } from "@nestjs/testing";
+import { UserController } from "./user.controller";
+import { User } from "./user.entity";
+import { UserService } from "./user.service";
+
+class NoErrorThrownError extends Error {}
+
+async function getError<TError>(call: () => unknown): Promise<TError> {
+	try {
+		await call();
+		throw new NoErrorThrownError();
+	} catch (error: unknown) {
+		return error as TError;
+	}
+}
+
+const mockUserData: User[] = [
+	{
+		id: 1,
+		auth_id: "auth0|6237590b647a36006ba3c6ea",
+		username: "johnsmith",
+		created: new Date(2022, 2, 30, 15, 32),
+		modified: new Date(2022, 3, 1, 24, 55)
+	},
+	{
+		id: 2,
+		auth_id: "auth0|6237590b6aaaaajfsdoalij2",
+		username: "testuser",
+		created: new Date(2022, 3, 1, 20, 44),
+		modified: new Date(2022, 3, 1, 20, 44)
+	}
+];
+
+describe("UserController", () => {
+	let service: UserService;
+	let controller: UserController;
+
+	beforeEach(async () => {
+		const module: TestingModule = await Test.createTestingModule({
+			controllers: [UserController],
+			providers: [
+				{
+					provide: UserService,
+					useValue: {
+						getUserByID: jest.fn().mockResolvedValue(mockUserData[0])
+					}
+				}
+			]
+		}).compile();
+
+		controller = module.get<UserController>(UserController);
+		service = module.get<UserService>(UserService);
+	});
+
+	it("should be defined", () => {
+		expect(controller).toBeDefined();
+	});
+
+	describe("/{id}", () => {
```

Comment:
I think since this isn't an e2e test it would be better to follow the test style we use for all other unit tests by using the method name rather than the route.